src/library_gl.js

```diff
@@ -1023,6 +1023,13 @@ var LibraryGL = {
 
   glCompressedTexImage2D__sig: 'viiiiiiii',
   glCompressedTexImage2D: function(target, level, internalFormat, width, height, border, imageSize, data) {
+#if USE_WEBGL2
+    if (GL.currentContext.version >= 2) { // WebGL 2 provides new garbage-free entry points to call to WebGL. Use those always when possible.
```

Comment:
I considered doing the stub functions, but it is more uncomfortable, since it would be a new function that would show up in the browser profilers tree, and these are among the hottest functions that are always at the top of profiling logs, so nice to have them look lean in the profiler. (Also I expect that an `if()` statement like this would be much faster than a function call, which interacts more heavily with JITs and other high level dynamic things). If we wanted to replace functions, we'd probably want to do that by replacing the whole top level functions `glCompressedTexImage2D` et al., but doing such a thing would need to be aware if one creates both WebGL 2 and WebGL 1 contexts and switches between them.